File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/NameDescriptionState.java
package net.nemerosa.ontrack.model.structure;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.beans.ConstructorProperties;

@EqualsAndHashCode(callSuper = false)
@Data
public class NameDescriptionState extends NameDescription {

    private final boolean disabled;

    @ConstructorProperties({"name", "description", "disabled"})
    public NameDescriptionState(String name, String description, boolean disabled) {
        super(name, description);
        this.disabled = disabled;
    }

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/NameDescriptionState.kt
package net.nemerosa.ontrack.model.structure

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import net.nemerosa.ontrack.model.structure.NameDescription.Companion.NAME
import net.nemerosa.ontrack.model.structure.NameDescription.Companion.NAME_MESSAGE_SUFFIX

data class NameDescriptionState(
        @get:NotNull(message = "The name is required.")
        @get:Pattern(regexp = NAME, message = "The name $NAME_MESSAGE_SUFFIX")
        val name: String,
        val description: String?,
        @get:JsonProperty("disabled")
        val isDisabled: Boolean
)
